hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/amrmproxy/FederationInterceptor.java

```diff
@@ -1434,6 +1449,17 @@ private void mergeAllocateResponses(AllocateResponse mergedResponse) {
         }
       }
     }
+    // When re-register RM, client may not cache the NMToken from register response.
+    // Here we pass these NMToken in allocate stage.
+    if (nmTokenMapFromRegisterSecondaryCluster.size() > 0) {
+      List<NMToken> duplicateNmToken = new ArrayList(nmTokenMapFromRegisterSecondaryCluster);
```

Comment:
If nmTokenMapFromRegisterSecondaryCluster is already a set, why is deduplication necessary?

Reply:
It is aim to avoid pass duplicated nmToken in next allocate.